### PR TITLE
Return Task in all cases from generated Script <Initialize> method; add Script<T> and ScriptState<T> to API

### PR DESCRIPTION
**Pull request by cston**
Return Task from generated Script &lt;Initialize&gt; method, even for non-async statements.
Add Script&lt;T&gt; and ScriptState&lt;T&gt; types (derived from Script and ScriptState) that make the submission return types explicit.

### CODE REVIEW

**Comment by cston** [comment]
@tmat, @mattwar

File: src/Scripting/CSharp/CSharpScript.cs [hunk 1]
```diff
@@ -3,45 +3,33 @@
...
-            return new CSharpScript(code, path, options, globalsType, returnType, builder, previous);
-        }
-
         /// <summary>
         /// Create a new C# script.
         /// <param name="code">The source code of the script.</param>
         /// <param name="options">The script options.</param>
         /// </summary>
-        public static Script Create(string code, ScriptOptions options)
+        public static Script<T> Create<T>(string code, ScriptOptions options)
```
Comment (mattwar) [inline src/Scripting/CSharp/CSharpScript.cs]: You should also have overloads that are not generic and just imply the type as object.

Comment (cston) [inline src/Scripting/CSharp/CSharpScript.cs]: Added.

Comment (cston) [inline src/Scripting/CSharp/CSharpScript.cs]: Changed generic overloads to non-generic and added one generic overload with all arguments in each case.

**Comment by mattwar** [comment]
:+1:

**Comment by tmat** [comment]
:+1:

**Comment by cston** [comment]
@tmat Ported Script.Initialize method change to VB

**Comment by cston** [comment]
Fixes #3719

**Pull request merged**